types/types.go

```diff
@@ -1,10 +1,13 @@
 package types
 
-import "github.com/gnolang/gno/tm2/pkg/std"
+import (
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
 
 // TxData contains the single block transaction,
 // along with the block information
 type TxData struct {
-	Tx       std.Tx `json:"tx"`
-	BlockNum uint64 `json:"blockNum"`
+	Tx        std.Tx `json:"tx"`
+	BlockNum  uint64 `json:"blockNum"`
+	Timestamp int64  `json:"bt"`
```

Comment:
IMO this could also simply be a `time.Time`; but `timestamp_ms` sounds good too.

It takes up "junk bytes"; but if space is a concern here we can just use a binary format, like amino, + compression.

I also wonder why `blockNum` is camelCase instead of snake_case, as is the norm in JSON.

Reply:
>I'm not using time.Time because, by default, marshaling/unmarshalling is not saving the timezone

???

RFC3339 stores the timestamp. The `Z` at the end of it indicates that it is UTC.

personally, I'd prefer if we used `blockTime` for consistency with `blockNum`; OR change them respectively to `bt` and `bn` if space really is a concern (this needs some kind of migration strategy, like a custom UnmarshalJSON which handles the case of blockNum and puts it in the appropriate field).

Reply:
> The Z at the end of JSON timestamps (RFC3339) indicates that it is UTC

Yep, I know that :). I meant that json library is always serializing Time types using the UTC timezone, losing some information on the process (might be interesting to have the original TZ to know where in the world approximately the transaction was created, as a dummy example). So I still don't see the point of using a Time type if it will use more space from one side compared with a timestamp and from the other it is not giving extra information, as the original timezone.